osu.Game/Screens/Edit/Compose/Components/RectangularPositionSnapGrid.cs

```diff
@@ -2,132 +2,51 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using osu.Framework.Graphics;
-using osu.Framework.Graphics.Containers;
-using osu.Framework.Graphics.Shapes;
-using osu.Framework.Layout;
-using osu.Framework.Utils;
+using osu.Framework.Bindables;
+using osu.Game.Utils;
 using osuTK;
 
 namespace osu.Game.Screens.Edit.Compose.Components
 {
-    public partial class RectangularPositionSnapGrid : CompositeDrawable
+    public partial class RectangularPositionSnapGrid : LinedPositionSnapGrid
     {
         /// <summary>
-        /// The position of the origin of this <see cref="RectangularPositionSnapGrid"/> in local coordinates.
+        /// The spacing between grid lines of this <see cref="RectangularPositionSnapGrid"/>.
         /// </summary>
-        public Vector2 StartPosition { get; }
-
-        private Vector2 spacing = Vector2.One;
+        public Bindable<Vector2> Spacing { get; } = new Bindable<Vector2>(Vector2.One);
 
         /// <summary>
-        /// The spacing between grid lines of this <see cref="RectangularPositionSnapGrid"/>.
+        /// The rotation in degrees of the grid lines of this <see cref="RectangularPositionSnapGrid"/>.
         /// </summary>
-        public Vector2 Spacing
-        {
-            get => spacing;
-            set
-            {
-                if (spacing.X <= 0 || spacing.Y <= 0)
-                    throw new ArgumentException("Grid spacing must be positive.");
-
-                spacing = value;
-                gridCache.Invalidate();
-            }
-        }
-
-        private readonly LayoutValue gridCache = new LayoutValue(Invalidation.RequiredParentSizeToFit);
+        public BindableFloat GridLineRotation { get; } = new BindableFloat();
 
-        public RectangularPositionSnapGrid(Vector2 startPosition)
+        public RectangularPositionSnapGrid()
         {
-            StartPosition = startPosition;
-
-            AddLayout(gridCache);
+            Spacing.BindValueChanged(_ => GridCache.Invalidate());
+            GridLineRotation.BindValueChanged(_ => GridCache.Invalidate());
         }
 
-        protected override void Update()
-        {
-            base.Update();
-
-            if (!gridCache.IsValid)
-            {
-                ClearInternal();
-
-                if (DrawWidth > 0 && DrawHeight > 0)
-                    createContent();
-
-                gridCache.Validate();
-            }
-        }
-
-        private void createContent()
+        protected override void CreateContent()
         {
             var drawSize = DrawSize;
+            var rot = Quaternion.FromAxisAngle(Vector3.UnitZ, MathHelper.DegreesToRadians(GridLineRotation.Value));
```

Comment:
using quaternions here feels a bit like shooting a fly with a cannon but sure I suppose

Reply:
I wanted to avoid calculating the sin and cos more than necessary. If there was a way to multiply a Matrix2 and a Vector2 I would've went with that.